pkg/kernel/mcp/iptables.go

```diff
@@ -0,0 +1,106 @@
+package kernel
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"github.com/ovn-kubernetes/ovn-kubernetes-mcp/pkg/kernel/types"
+)
+
+// GetIptables MCP handler for iptables operations.
+// GetIptables retrieves iptables/ip6tables rules from a Kubernetes node.
+// Automatically detects IPv6 and uses ip6tables when needed.
+func (s *MCPServer) GetIptables(ctx context.Context, req *mcp.CallToolRequest, in types.ListIPTablesParams) (*mcp.CallToolResult, types.Result, error) {
+	iptablesCliAvailable, err := s.UtilityExists(ctx, req, in.Node, in.Image, "iptables")
+	if !iptablesCliAvailable {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: mentioned image does not have iptables utility: %w", err)
+	}
+
+	if err := ValidateTableName(in.Table); err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+	}
+	if err := ValidateIptablesCommand(in.Command); err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+	}
+	if in.FilterParameters != "" {
+		if err := validateFilterParameters(in.FilterParameters); err != nil {
+			return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+		}
+	}
+
+	cmd := newCommand(iptablesCommand(in.FilterParameters))
+	// Defaults to 'filter' table when not specified
+	cmd.addIf(in.Table == "", "-t", "filter")
+	cmd.addIfNotEmpty(in.Table, "-t", in.Table)
+	// Defaults to -L (list) when command not specified
+	cmd.addIf(in.Command == "", "-L")
+	cmd.addIfNotEmpty(in.Command, in.Command)
+	// FilterParameters are invalid with -S command
+	cmd.addIf(in.FilterParameters != "" && in.Command != "-S", strings.Fields(in.FilterParameters)...)
+
+	stdout, err := s.executeCommand(ctx, req, in.Node, in.Image, cmd.build())
+	if err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting list of iptables rules: %w", err)
+	}
+	stdout = limitOutputLines(stdout, in.MaxLines)
+	return nil, types.Result{Data: stdout}, nil
+}
+
+// iptablesCommand determines whether to use iptables or ip6tables.
+func iptablesCommand(filterParameters string) string {
+	for _, item := range strings.Split(filterParameters, " ") {
+		// Use of ip6tables CLI is required while either --ipv6 or -6 flag is mentioned
+		if item == "--ipv6" || (strings.Contains(item, "-") && strings.Contains(item, "6")) {
+			return "ip6tables"
+		}
+	}
+	return "iptables"
+}
```

Comment:
Let's add some unit tests for `iptablesCommand` using different values for filterParameters.